.github/workflows/test-upgrade.yaml

```diff
@@ -0,0 +1,135 @@
+---
+name: Kuadrant Upgrade Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      kuadrantStartVersion:
+        description: Kuadrant start version
+        required: true
+        type: string
+      clusterServer:
+        description: Cluster server URL
+        required: false
+        type: string
+      clusterToken:
+        description: Cluster Server Bearer Token
+        required: false
+        type: string
+      kuadrantNamespace:
+        description: Namespace where Kuadrant is installed
+        required: false
+        default: kuadrant-system
+        type: string
+jobs:
+  helm-charts-upgrade-test:
+    runs-on: ubuntu-latest
+    name: Helm Charts Upgrade Test
+    if: ${{ github.event_name == 'workflow_dispatch' }}
```

Comment:
not needed since the only event listener is `workflow_dispatch` https://github.com/Kuadrant/kuadrant-operator/pull/1156/files#diff-e3c60a7e344a4151f98dbfd60770be22bc6169d9c3c2cb7c4748529a3e9a6333R4-R5